src/prepare_dlc_dev_environment.py

```diff
@@ -525,6 +545,113 @@ def create_docker_file(docker_file_path):
         LOGGER.warning(f"WARNING: Failed to create {docker_file_path}. Error: {e}")
 
 
+def override_existing_buildspec(buildspec_path, override_tags):
+    """
+    Override the autopatch_build and build_tag_override tags in an existing buildspec file.
+    """
+    full_path = validate_buildspec_path(buildspec_path)
+    if not full_path:
+        return
+
+    with open(full_path, "r") as file:
+        content = file.readlines()
+
+    autopatch_build_found = False
+    build_tag_override_found = False
+    updated_content = []
+
+    for line in content:
+        if line.strip().startswith("autopatch_build"):
+            autopatch_build_found = True
+            updated_line = handle_autopatch_build_line(line, override_tags)
+        elif line.strip().startswith("# build_tag_override:"):
+            build_tag_override_found = True
+            updated_line = handle_build_tag_override_line(line, override_tags)
+        else:
+            updated_line = line
+
+        updated_content.append(updated_line)
+
+    if not autopatch_build_found and override_tags:
+        LOGGER.warning(f"WARNING: No autopatch_build tag found in {buildspec_path}")
+
+    if not build_tag_override_found and override_tags:
+        LOGGER.warning(f"WARNING: No build_tag_override tag found in {buildspec_path}")
+
+    if autopatch_build_found or build_tag_override_found:
```

Comment:
The only requirement here is that build tag overrides is true, which may be implicit if you return in the conditions above. Ideally we don't want a scenario where we comment autopatch but don't do anything with the overrides, as that will be unexpected behavior from a user perspective